crates/debugger/src/tui/draw.rs

```diff
@@ -356,7 +358,20 @@ impl DebuggerContext<'_> {
                 let pc_ic_map = if is_create { create_map } else { rt_map };
                 let ic = pc_ic_map.get(pc)?;
                 let source_element = source_map.swap_remove(ic);
-                (*file_id == source_element.index?).then_some((source_element, source_code))
+                // if the source element has an index, find the sourcemap for that index
+                source_element
+                    .index
+                    .and_then(|index| 
+                    // if index matches current file_id, return current source code
+                    (index == file_id).then_some((source_element.clone(), source_code)))
```

Comment:
```suggestion
                    (index == file_id).then(|| (source_element.clone(), source_code)))
```

Reply:
Changed – curious, is there a functional difference between the two?